src/allmydata/web/status.py

```diff
@@ -1128,49 +1129,92 @@ def childFactory(self, ctx, name):
                 if s.get_counter() == count:
                     return RetrieveStatusPage(s)
 
-
-class HelperStatus(MultiFormatPage):
-    docFactory = getxmlfile("helper.xhtml")
+# Render "/helper_status" page.
+class HelperStatus(MultiFormatResource):
 
     def __init__(self, helper):
-        rend.Page.__init__(self, helper)
-        self.helper = helper
+        super(HelperStatus, self).__init__()
+        self._helper = helper
 
-    def data_helper_stats(self, ctx, data):
-        return self.helper.get_stats()
+    def render_HTML(self, req):
+        return renderElement(req, HelperStatusElement(self._helper))
 
     def render_JSON(self, req):
         req.setHeader("content-type", "text/plain")
-        if self.helper:
-            stats = self.helper.get_stats()
+        if self._helper:
+            stats = self._helper.get_stats()
             return json.dumps(stats, indent=1) + "\n"
         return json.dumps({}) + "\n"
 
-    def render_active_uploads(self, ctx, data):
-        return data["chk_upload_helper.active_uploads"]
+class HelperStatusElement(Element):
 
-    def render_incoming(self, ctx, data):
-        return "%d bytes in %d files" % (data["chk_upload_helper.incoming_size"],
-                                         data["chk_upload_helper.incoming_count"])
+    loader = XMLFile(FilePath(__file__).sibling("helper.xhtml"))
 
-    def render_encoding(self, ctx, data):
-        return "%d bytes in %d files" % (data["chk_upload_helper.encoding_size"],
-                                         data["chk_upload_helper.encoding_count"])
+    def __init__(self, helper):
+        """
+        :param _allmydata.immutable.offloaded.Helper helper
+        """
```

Comment:
this needs at least a trailing `:` I think

Reply:
Eeps, I got careless!  Fixed in f02501e.  Thank you for catching this.